### PR TITLE
Add "Upgrading from Godot 4.1 to Godot 4.2" page

### PR DESCRIPTION
**Pull request by raulsntos**
Adds the migration guide for 4.2, just like we did for 4.1 (https://github.com/godotengine/godot-docs/pull/7611).

Last time we didn't include GDExtension compatibility because 4.1 completely broke compatibility, we added a warning at the top of the page instead. I'm not sure how we want to handle GDExtension compatibility this time, the @godotengine/gdextension team can suggest changes or make a follow-up PR.

Area maintainers should review the changes listed for their area. Also feel free to suggest any addition that I might have missed or make follow-up PRs.

### CODE REVIEW

File: tutorials/migrating/upgrading_to_godot_4.2.rst [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+This page intends to cover everything you need to pay attention to when migrating
+your project.
+
+Breaking changes
+----------------
+
+If you are migrating from 4.1 to 4.2, the breaking changes listed here might
+affect you. Changes are grouped by areas/systems.
+
+This article indicates whether each breaking change affects GDScript and whether
```
Comment (Calinou) [inline tutorials/migrating/upgrading_to_godot_4.2.rst:16]: Add section on Mesh format compatibility:

```suggestion
affect you. Changes are grouped by areas/systems.

.. warning::

    The Mesh resource format has changed in 4.2 to allow for
    `vertex and attribute compression <https://github.com/godotengine/godot/pull/81138>`__. This allows for improved rendering performance, especially
    on platforms constrained by memory bandwidth such as mobile.
	
    It is still possible to load the Godot 4.0-4.1 Mesh formats, but it is
    **not** possible to load the Godot 4.2 Mesh format in prior Godot versions.
    When opening a project, you will be presented with an upgrade dialog that has two options:
	
    - **Restart & Upgrade:** This prevents downgrading the project to a Godot version
      older than 4.2. Set up a version control system and push your changes
      *before* choosing this option!
    - **Upgrade Only:** Upgrades the mesh format in-memory without writing it
      to disk. This allows downgrading the project to a Godot version older than 4.2
      if you need to do so in the future. The downside is that loading the project
      will be slower every time as the mesh format needs to be upgraded every time
      the project is loaded. These increased loading times will also affect the
      exported project. The number and complexity of Mesh resources determines
      how much loading times are be affected.

This article indicates whether each breaking change affects GDScript and whether
```

File: tutorials/migrating/upgrading_to_godot_4.2.rst [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+      project and saves the result to disk. Once chosen, this option prevents
+      downgrading the project to a Godot version prior to 4.2. Set up a
+      version control system and push your changes *before* choosing this option!
+    - **Upgrade Only:** Upgrades the mesh format in-memory without writing it
+      to disk. This allows downgrading the project to a Godot version older than 4.2
+      if you need to do so in the future. The downside is that loading the project
+      will be slower every time as the mesh format needs to be upgraded every time
+      the project is loaded. These increased loading times will also affect the
+      exported project. The number and complexity of Mesh resources determines
+      how much loading times are be affected.
```
Comment (Calinou) [inline tutorials/migrating/upgrading_to_godot_4.2.rst:38]:
```suggestion
.. warning::

    The :ref:`class_Mesh` resource format has changed in 4.2 to allow for
    `vertex and attribute compression <https://github.com/godotengine/godot/pull/81138>`__.
    This allows for improved rendering performance, especially on platforms
    constrained by memory bandwidth such as mobile.

    It is still possible to load the Godot 4.0-4.1 Mesh formats, but it is
    **not** possible to load the Godot 4.2 Mesh format in prior Godot versions.
    When opening a Godot project made with a version prior to 4.2, you may
    be presented with an upgrade dialog that offers two options:

    - **Restart & Upgrade:** Upgrades the mesh format for all meshes in the
      project and saves the result to disk. Once chosen, this option prevents
      downgrading the project to a Godot version prior to 4.2. Set up a
      version control system and push your changes *before* choosing this option!
    - **Upgrade Only:** Upgrades the mesh format in-memory without writing it
      to disk. This allows downgrading the project to a Godot version older than 4.2
      if you need to do so in the future. The downside is that loading the project
      will be slower every time as the mesh format needs to be upgraded every time
      the project is loaded. These increased loading times will also affect the
      exported project. The number and complexity of Mesh resources determines
      how much loading times are be affected.
    
    If this dialog doesn't appear, use **Project > Tools > Upgrade Mesh Surfaces…**
    at the top of the editor.
```

Comment (Cerno-b) [inline tutorials/migrating/upgrading_to_godot_4.2.rst:38]: Typo "are be"

File: tutorials/migrating/upgrading_to_godot_4.2.rst [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+Method ``get_zoom_hbox`` renamed to ``get_menu_hbox``                                                                     |❌|                 |✔️ with compat|      |✔️ with compat|      `GH-79308`_
+Property ``snap_distance`` renamed to ``snapping_distance``                                                               |❌|                 |✔️ with compat|      |✔️ with compat|      `GH-79308`_
+Property ``use_snap`` renamed to ``snapping_enabled``                                                                     |❌|                 |✔️ with compat|      |✔️ with compat|      `GH-79308`_
+**GraphNode**
+Property ``comment`` removed                                                                                              |❌|                 |❌|                  |❌|                  `GH-79307`_
+Signal ``close_request`` renamed to``delete_request`` and moved to base class ``GraphElement``                            |❌|                 |✔️ with compat|      |✔️ with compat|      `GH-79311`_
```
Comment (Cerno-b) [inline tutorials/migrating/upgrading_to_godot_4.2.rst:171]: space missing before ``delete_request``

**Review by Cerno-b — CHANGES_REQUESTED** [review]
I just did a language review, I have no clue about content accuracy ;)

File: tutorials/migrating/upgrading_to_godot_4.2.rst [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+Change                                                                                                                                GDScript Compatible  C# Binary Compatible  C# Source Compatible  Introduced
+====================================================================================================================================  ===================  ====================  ====================  ===========
+**Font**
+Method ``set_fallbacks`` replaced with ``fallbacks`` property                                                                         |✔️|                 |❌|                  |❌|                  `GH-78266`_
+Method ``get_fallbacks`` replaced with ``fallbacks`` property                                                                         |✔️|                 |❌|                  |❌|                  `GH-78266`_
+Method ``find_variation`` adds new ``spacing_top``, ``spacing_bottom``, ``spacing_space`` and ``spacing_glyph`` optional parameters   |✔️|                 |✔️ with compat|      |✔️|                  `GH-80954`_
```
Comment (AThousandShips) [inline tutorials/migrating/upgrading_to_godot_4.2.rst:155]:
```suggestion
Method ``find_variation`` adds new ``spacing_top``, ``spacing_bottom``, ``spacing_space``, and ``spacing_glyph`` optional parameters   |✔️|                 |✔️ with compat|      |✔️|                  `GH-80954`_
```
Please use [Oxford comma](https://docs.godotengine.org/en/latest/contributing/documentation/docs_writing_guidelines.html#use-the-oxford-comma-to-enumerate-anything).

File: tutorials/migrating/upgrading_to_godot_4.2.rst [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+========================================================================================================================  ===================  ====================  ====================  ===========
+**PopupMenu**
+Method ``add_icon_shortcut`` adds a new ``allow_echo`` optional parameter                                                 |✔️|                 |✔️ with compat|      |✔️ with compat|      `GH-36493`_
+Method ``add_shortcut`` adds a new ``allow_echo`` optional parameter                                                      |✔️|                 |✔️ with compat|      |✔️ with compat|      `GH-36493`_
+Method ``clear`` adds a new ``free_submenus`` optional parameter                                                          |✔️|                 |✔️ with compat|      |✔️ with compat|      `GH-79965`_
+**RichTextLabel**
+Method ``add_image`` adds new ``key``, ``pad``, ``tooltip`` and ``size_in_percent`` optional parameters                   |✔️|                 |✔️ with compat|      |✔️|                  `GH-80410`_
```
Comment (AThousandShips) [inline tutorials/migrating/upgrading_to_godot_4.2.rst:121]:
```suggestion
Method ``add_image`` adds new ``key``, ``pad``, ``tooltip``, and ``size_in_percent`` optional parameters                   |✔️|                 |✔️ with compat|      |✔️|                  `GH-80410`_
```

**Review by skyace65 — APPROVED** [review]

**Review by mhilbrunner — APPROVED** [review]

**Comment by mhilbrunner** [comment]
Thank you! Merged.

**Pull request merged**